File: output/Llama-2-100/avatar/Java/Python/codeforces_302_A.py
def codeforces_302_A(n, k): Scanner = input Scanner n = int(n) k = int(k) arr = [int(i) for i in range(n)] o = 0 e = 0 for i in range(n): if arr[i] == 1: o += 1 else: e += 1 for i in range(k): l = int(input()) r = int(input()) if (r - l + 1) % 2 == 1: print("0\n") else: if (r - l + 1) // 2 <= o and (r - l + 1) // 2 <= e: print("1\n") else: print("0\n")

File: output/Llama-2-100/avatar/Python/Java/codeforces_302_A.java
import java.util.Scanner;

public class codeforces_302_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int sa = scanner.nextInt();
        int minSa = Math.min(n - sa, sa);
        String[] ss = new String[m];
        for (int i = 0; i < m; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            b -= a;
            if (b % 2 == 0 && b <= minSa << 1) {
                ss[i] = "1\n";
            } else {
                ss[i] = "0\n";
            }
        }
        System.out.println(String.join("", ss));
    }
}


